### PR TITLE
Use Maths lib on determing amount to repay, remove unused debtToPay c…

### PR DESCRIPTION
**Pull request by grandizzy**
…alculation from Buckets lib

### CODE REVIEW

File: src/ERC20Pool.sol [hunk 1]
```diff
@@ -353,19 +353,7 @@ contract ERC20Pool is IPool, Clone {
...
-        uint256 curLup = lup;
-        if (curLup == 0) {
-            curLup = hpb;
-        }
-
-        // TODO: make value explicit for use in comparison operator against collateralDeposited below
-        uint256 encumberedBorrowerCollateral;
-        if (borrower.debt != 0) {
-            encumberedBorrowerCollateral = getEncumberedCollateral(borrower.debt);
-        }
```
Comment (EdNoepel) [inline src/ERC20Pool.sol:365]: Indeed; this block wasn't needed.  Line 360 adds existisng `borrower.debt` with new `_amount` to calculate overall collateralization.

File: src/ERC20Pool.sol [hunk 2]
```diff
@@ -404,17 +392,10 @@ contract ERC20Pool is IPool, Clone {
...
-            amount = _maxAmount;
-        }
-
-        uint256 debtToPay;
-        (lup, debtToPay) = _buckets.repay(amount, lup, inflatorSnapshot);
+        uint256 amount = Maths.min(_maxAmount, borrower.debt);
+        lup = _buckets.repay(amount, lup, inflatorSnapshot);
 
-        borrower.debt -= Maths.min(borrower.debt, amount);
+        borrower.debt -= amount;
```
Comment (EdNoepel) [inline src/ERC20Pool.sol:398]: Curious whether you hit an underflow issue due to pending debt tie-out issues.

File: src/libraries/Buckets.sol [hunk 3]
```diff
@@ -168,7 +161,7 @@ library Buckets {
             _lup = curLup.price;
         }
 
-        return (_lup, loanCost);
+        return _lup;
```
Comment (EdNoepel) [inline src/libraries/Buckets.sol:164]: Agreed; `loanCost` should always equal `_amount`.

**Review by EdNoepel — APPROVED** [review]
Appreciate the improvement; I'm looking forward to running a new real-world with this.

File: src/ERC20Pool.sol (hunk 2 above)
Comment (grandizzy) [reply src/ERC20Pool.sol:398]: Nope, that's not possible as `amount` is the min of `maxAmount` and `borrower.debt` at line 395

**Pull request merged**